File: ts/lcp/02.ts
function fraction(cont: number[]): number[] {
  let i: number = cont.length - 1
  let n: number = 1, m: number = cont[i--]
  while (i >= 0) {
    let tmp = cont[i] * m + n
    n = m
    m = tmp
    i -= 1
  }
  return [m, n]
}

File: py3/lcp/02.py
from typing import List

class Solution:
    def fraction(self, cont: List[int]) -> List[int]:
        i = len(cont) - 1
        n, m = 1, cont[i]
        i -= 1
        while i >= 0:
            tmp = cont[i] * m + n
            n = m
            m = tmp
            i -= 1

        return [m, n]